tests/test_cell_statistics.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import annotations
+
+from unittest.mock import patch
+
+import numpy as np
+import pytest
+
+import napari
+
+import epitools._sample_data
+from epitools.analysis import calculate_cell_statistics
+
+
+@pytest.fixture(scope="function")
+def projected_image() -> napari.layers.Image:
+    data, kwargs, layer_type = epitools._sample_data.load_projected_data()[0]
+    return napari.layers.Image(data, **kwargs)
+
+
+@pytest.fixture(scope="function")
+def seeds_and_labels(
+    make_napari_viewer,
+) -> tuple[napari.layers.Points, napari.layers.Labels]:
+    seeds_layer_data, labels_layer_data = epitools._sample_data.load_segmented_data()
+    seeds_data, seeds_kwargs, _ = seeds_layer_data
+    labels_data, labels_kwargs, _ = labels_layer_data
+    return (
+        napari.layers.Points(seeds_data, **seeds_kwargs),
+        napari.layers.Labels(labels_data, **labels_kwargs),
+    )
+
+
+@pytest.fixture(scope="function")
+def viewer_with_image(
+    make_napari_viewer,
+    projected_image,
+    seeds_and_labels,
+) -> napari.Viewer:
+    seeds, labels = seeds_and_labels
+    viewer = make_napari_viewer()
+
+    viewer.add_layer(projected_image)
+    viewer.add_layer(labels)
+    viewer.add_layer(seeds)
+
+    return viewer
+
+
+def test_add_cell_statistics_widget(make_napari_viewer):
+    """Checks that the cell statistics widget can be added inside a dock widget."""
+
+    viewer = make_napari_viewer()
+    num_dw = len(list(viewer.window._dock_widgets))
+    viewer.window.add_plugin_dock_widget(
+        plugin_name="epitools",
+        widget_name="Cell statistics",
+    )
+
+    assert len(list(viewer.window._dock_widgets)) == num_dw + 1
+
+
+def test_segmentation_widget_run_button(
+    viewer_with_image,
+    seeds_and_labels,
+):
+    """
+    Check that pressing the 'Run' button calculates cell statistics
+    using the selected image and labels layers, and that the labels layer
+    has it's features set to the calculated cell statistics.
+    """
+
+    dock_widget, container = viewer_with_image.window.add_plugin_dock_widget(
+        plugin_name="epitools",
+        widget_name="Cell statistics",
+    )
+
+    reference_seeds, reference_labels = seeds_and_labels
+    reference_stats = reference_labels.metadata["cell_statistics"]
+
+    # Check cells layer has no features (i.e. no cell stats)
+    # before pressing the 'Run' button
+    image_layer, cells_layer, seeds_layer = viewer_with_image.layers
+    assert not viewer_with_image.layers["Cells"].features.size
```

Comment:
So basically we're checking that the size is empty? Can you do this in a more intuitive way? The `assert not` is a bit confusing. Even `.size == 0` is more readable.

Reply:
> So basically we're checking that the size is empty?

yeah exactly, I've changed it so we compare `.size == 0`